.github/workflows/run_tests.yml

```diff
@@ -23,15 +23,31 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install ./acasclient
-      - name: Set ACAS_TAG to latest release
+      # The following steps are all to figure out the right ACAS_REF (git ref) and ACAS_TAG (docker tag) of ACAS to use
+      # If main or release branch, run tests on matching release branch of ACAS
+      - name: Set ACAS_REF to the current branch ${{ github.ref }}
         run: |
-          echo "ACAS_TAG=$(curl -s https://api.github.com/repos/mcneilco/acas/releases/latest | jq -r '.tag_name')" >> $GITHUB_ENV
+          echo "ACAS_REF=${{ github.ref }}" >> $GITHUB_ENV
+        if: github.event_name == 'push'
+      # If a PR, run tests on ACAS branch matching destination of PR
+      - name: Set ACAS_REF to the PR destination branch ${{ github.base_ref }}
+        run: |
+          echo "ACAS_REF=${{ github.base_ref }}" >> $GITHUB_ENV
+        if: github.event_name == 'pull_request'
+      - name: Set ACAS_TAG to the docker escaped version of ${{ env.ACAS_REF }}
+        run: |
+          echo "ACAS_TAG=$(echo ${{ env.ACAS_REF }} | sed 's/\//-/g')" >> $GITHUB_ENV
+      - name: Override ACAS_REF and ACAS_TAG if branch is "main"
+        run: |
+          echo "ACAS_REF=master" >> $GITHUB_ENV
```

Comment:
Instead of hardcoding master here you could do:

```
curl -s https://api.github.com/repos/mcneilco/acas | jq -r '.default_branch'
```

Reply:
Hm.... do we actually want the default branch though? I think we already cover testing against the latest release branch via the corresponding release branch of acasclient.

My unspoken assumption here is that we'd upmerge `release/1.13.7` -> `release/2022.1.x` -> `master` in acas, and `release/1.13.7` -> `release/2022.1.x` -> `main` in acasclient, so I was hardcoding a correspondence between `master` and `main`